test/unit/org/apache/cassandra/cql3/validation/operations/SelectTest.java

```diff
@@ -341,8 +344,11 @@ public void testSetContainsWithIndex() throws Throwable
             assertInvalidMessage("Unsupported unset value for column categories",
                                  "SELECT * FROM %s WHERE account = ? AND id = ? AND categories CONTAINS ?", "test", 5, unset());
 
-            assertInvalidMessage(StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE,
-                                 "SELECT * FROM %s WHERE account = ? AND categories CONTAINS ? AND categories CONTAINS ?", "xyz", "lmn", "notPresent");
+            if (DatabaseDescriptor.getDefaultSecondaryIndex().equals(CassandraIndex.NAME))
```

Comment:
I wonder if it would read better to move this that is repeated so much into a method in CQLTester with some javadoc explanation